CMakeLists.txt

```diff
@@ -63,8 +63,16 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
-
-find_package(Z3 4.8.0)
+set(LATEST_Z3_VERSION "4.8.12")
+if (TESTS AND NOT EMSCRIPTEN)
+  find_package(Z3 ${LATEST_Z3_VERSION} REQUIRED)
```

Comment:
I'd use `EXACT` here because tests will likely fail on a newer version too when it's released.

Reply:
I'll fix it